lib/node_modules/@stdlib/math/base/assert/is-nonnegative-integer/src/addon.c

```diff
@@ -0,0 +1,23 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/assert/is_nonnegative_integer.h"
+#include "stdlib/math/base/napi/unary.h"
+
+// cppcheck-suppress shadowFunction
+STDLIB_MATH_BASE_NAPI_MODULE_D_D( stdlib_base_is_nonnegative_integer )
```

Comment:
This isn't correct, as `D_D` corresponds to double-in and double-out. We should probably add `D_X` to `math/base/napi/unary` as a separate PR and then update this PR accordingly.

Reply:
Left a follow-up comment below. The other alternative is to manually perform argument munging.